### ISSUE TITLE
Provide additional options for spot-perpetual-arbitrage to reduce liquidation risks

### ISSUE THREAD

**Issue by tedwardd** [CoinAlpha/hummingbot#3630]
**Feature Suggestion**
// A succinct description of **how you want to achieve the desired behavior**.
// If you have looked at the code and know exactly what code changes are needed then please _consider submitting a pull request_ instead.
Provide an option in the spot-perpetual-arbitrage strategy to select between cross or isolated margin as well an option to define the amount of margin to use for isolated orders, similar to how we currently specify the order amount.

**Impact**
// A succinct description of **why you want the _desired behavior_** specified above.
Currently, users of the spot-perpetual-arbitrage strategy run the risk of being liquidated because the bot places derivative orders using isolated margin. This is great for simple risk management but it exposes the trader to the risk of liquidation should positions stay open long enough to experience a somewhat large price movement.

**Additional context**
Perhaps this is an opportunity to improve documentation around this strategy, however, my understanding of spot-perpetual-arbitrage is that you are taking out two orders, one in spot and one on the derivative markets. These orders are opposite of one another (if spot is long, perpetual is short, and vice-versa), the spot order acting as the collateral for the perpetual contract. Eventually the prices of the two will converge below the convergence threshold configured for the bot and both deal will be closed. It is unclear, however, what happens when the perpetual position is liquidated before the prices converge.

Maybe this is where my understanding of the strategy needs improvement but It would seem to me that it would make sense to provide advanced options on spot-perpetual-arbitrage bots to either use cross leverage or a custom amount of margin to mitigate the possibility of this happening. If the risk of liquidation is not of concern, then I feel as though the documentation around the strategy should be clarified to indicate this.

**Comment by hkingofsaga** [comment]
Yeah, If this strategy don’t deal with liquidation risks, we can’t use it at all!
